### ISSUE TITLE
Android: build failure with ndk 27-beta1 due to use of deprecated ALooper_pollAll

### ISSUE THREAD

**Issue by madebr** [libsdl-org/SDL#9792]
Building SDL3 with ndk 27-beta1 fails with the following error:
```
/home/maarten/projects/SDL/src/sensor/android/SDL_androidsensor.c:73:13: error: 'ALooper_pollAll' is unavailable: obsoleted in Android 1
   73 |         if (ALooper_pollAll(-1, NULL, &events, (void **)&source) == LOOPER_ID_USER) {
      |             ^
```
Configuring with `-DCMAKE_C_FLAGS="-Wno-error=deprecated" -DSDL_WERROR=OFF` does *not* turn the error in a warning.

[ALooper_pollAll documentation](https://developer.android.com/ndk/reference/group/looper#alooper_pollall)

**Comment by jsm174** [comment]
Looks like github runners are removing NDK 25 very soon: https://github.com/actions/runner-images/issues/10342. While trying to compile 2.30.6, I started seeing this issue:

```
./src/sensor/android/SDL_androidsensor.c:164:9: error: 'ALooper_pollAll' is unavailable: obsoleted in Android 1 - ALooper_pollAll may ignore wakes. Use ALooper_pollOnce instead. See The API documentation for more information
  164 |     if (ALooper_pollAll(0, NULL, &events, (void **)&source) == LOOPER_ID_USER) {
      |         ^
/usr/local/lib/android/sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/android/looper.h:228:5: note: 'ALooper_pollAll' has been explicitly marked unavailable here
  228 | int ALooper_pollAll(int timeoutMillis, int* outFd, int* outEvents, void** outData)
      |     ^
1 warning and 1 error generated.

```

**Comment by slouken** [comment]
This is fixed for the next release in 4222cebef, thanks!
